segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/EntryIteratorState.java

```diff
@@ -0,0 +1,137 @@
+     * Creates a new instance of the EntryIteratorState class.
+     *
+     * @param position The position of the TableEntry
+     * @param reachedEnd If the Entry is at the end of the segment (more recently appended).
+     * @param shouldClear Marks if the client should clear their state (provided start position has been truncated).
+     * @param deletionRecord The Entry read is marked for deletion.
+     */
+    EntryIteratorState(long position, boolean reachedEnd, boolean shouldClear, boolean deletionRecord) {
+        Preconditions.checkArgument(isValid(position), "Position must be at least 0 (a non-negative integer).");
+        this.position = position;
+        this.reachedEnd = reachedEnd;
+        this.shouldClear = shouldClear;
+        this.deletionRecord = deletionRecord;
+    }
+
+    boolean isValid(long position) {
+        return position >= 0;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("Position = %s Reached End = %s Should Clear = %s", this.position, this.reachedEnd, this.shouldClear);
+    }
+
+    //region Serialization
+
+    /**
+     * Creates a new instance of the IteratorState class from the given array.
+     *
+     * @param data A byte array containing the serialization of an IteratorState. This must have been generated using
+     *             {@link #serialize()}.
+     * @return As new instance of the IteratorState class.
+     */
+    @SneakyThrows(IOException.class)
+    public static EntryIteratorState deserialize(byte[] data) {
+        return SERIALIZER.deserialize(data);
+    }
+
+    /**
+     * Serializes this IteratorState instance into an {@link ArrayView}.
+     *
+     * @return The {@link ArrayView} that was used for serialization.
+     */
+    @SneakyThrows(IOException.class)
+    public ArrayView serialize() {
+        return SERIALIZER.serialize(this);
+    }
+
+    private static class IteratorStateBuilder implements ObjectBuilder<EntryIteratorState> {
+        private long position;
+        private boolean reachedEnd;
+        private boolean shouldClear;
+        private boolean deletionRecord;
+
+        @Override
+        public EntryIteratorState build() {
+            return new EntryIteratorState(position, reachedEnd, shouldClear, deletionRecord);
+        }
+    }
+
+    private static class Serializer extends VersionedSerializer.WithBuilder<EntryIteratorState, IteratorStateBuilder> {
+        @Override
+        protected IteratorStateBuilder newBuilder() {
+            return new IteratorStateBuilder();
+        }
+
+        @Override
+        protected byte getWriteVersion() {
+            return 0;
+        }
+
+        @Override
+        protected void declareVersions() {
+            version(0).revision(0, this::write00, this::read00);
+        }
+
+        private void read00(RevisionDataInput revisionDataInput, IteratorStateBuilder builder) throws IOException {
+            builder.position = revisionDataInput.readCompactLong();
```

Comment:
Just a heads up that CompactLong can only serialize up to 2^62 (and no negatives). This sounds OK for now, since we do not expect negative offsets and 2^62 is still a huge number.